
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Development Site of the GA4GH Beacon community">
      
      
        <meta name="author" content="Michael Baudis">
      
      
      
      
      
      <link rel="icon" href="../../img/ga4gh_circle.ico">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.14">
    
    
      
        <title>Request Parameter Definitions - GA4GH Beacon Genomic Variation Query Standards</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.342714a4.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      
  
  
    
    
  
  
  <style>:root{--md-admonition-icon--bug:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%20384%20512%22%3E%3C%21--%21%20Font%20Awesome%20Free%206.7.2%20by%20%40fontawesome%20-%20https%3A//fontawesome.com%20License%20-%20https%3A//fontawesome.com/license/free%20%28Icons%3A%20CC%20BY%204.0%2C%20Fonts%3A%20SIL%20OFL%201.1%2C%20Code%3A%20MIT%20License%29%20Copyright%202024%20Fonticons%2C%20Inc.--%3E%3Cpath%20d%3D%22M0%2024C0%2010.7%2010.7%200%2024%200h336c13.3%200%2024%2010.7%2024%2024s-10.7%2024-24%2024h-8v19c0%2040.3-16%2079-44.5%20107.5L225.9%20256l81.5%2081.5C336%20366%20352%20404.7%20352%20445v19h8c13.3%200%2024%2010.7%2024%2024s-10.7%2024-24%2024H24c-13.3%200-24-10.7-24-24s10.7-24%2024-24h8v-19c0-40.3%2016-79%2044.5-107.5l81.6-81.5-81.6-81.5C48%20146%2032%20107.3%2032%2067V48h-8C10.7%2048%200%2037.3%200%2024m110.5%20347.5c-3.9%203.9-7.5%208.1-10.7%2012.5h184.4c-3.2-4.4-6.8-8.6-10.7-12.5L192%20289.9l-81.5%2081.5zM284.2%20128C297%20110.4%20304%2089%20304%2067V48H80v19c0%2022.1%207%2043.4%2019.8%2061z%22/%3E%3C/svg%3E');}</style>



    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../css/theme_overrides.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="teal" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#adjacencyaccession" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="GA4GH Beacon Genomic Variation Query Standards" class="md-header__button md-logo" aria-label="GA4GH Beacon Genomic Variation Query Standards" data-md-component="logo">
      
  <img src="../../img/GA-logo.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            GA4GH Beacon Genomic Variation Query Standards
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Request Parameter Definitions
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91s2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/ga4gh-beacon/variant-query-types" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 480 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M186.1 328.7c0 20.9-10.9 55.1-36.7 55.1s-36.7-34.2-36.7-55.1 10.9-55.1 36.7-55.1 36.7 34.2 36.7 55.1M480 278.2c0 31.9-3.2 65.7-17.5 95-37.9 76.6-142.1 74.8-216.7 74.8-75.8 0-186.2 2.7-225.6-74.8-14.6-29-20.2-63.1-20.2-95 0-41.9 13.9-81.5 41.5-113.6-5.2-15.8-7.7-32.4-7.7-48.8 0-21.5 4.9-32.3 14.6-51.8 45.3 0 74.3 9 108.8 36 29-6.9 58.8-10 88.7-10 27 0 54.2 2.9 80.4 9.2 34-26.7 63-35.2 107.8-35.2 9.8 19.5 14.6 30.3 14.6 51.8 0 16.4-2.6 32.7-7.7 48.2 27.5 32.4 39 72.3 39 114.2m-64.3 50.5c0-43.9-26.7-82.6-73.5-82.6-18.9 0-37 3.4-56 6-14.9 2.3-29.8 3.2-45.1 3.2-15.2 0-30.1-.9-45.1-3.2-18.7-2.6-37-6-56-6-46.8 0-73.5 38.7-73.5 82.6 0 87.8 80.4 101.3 150.4 101.3h48.2c70.3 0 150.6-13.4 150.6-101.3m-82.6-55.1c-25.8 0-36.7 34.2-36.7 55.1s10.9 55.1 36.7 55.1 36.7-34.2 36.7-55.1-10.9-55.1-36.7-55.1"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="GA4GH Beacon Genomic Variation Query Standards" class="md-nav__button md-logo" aria-label="GA4GH Beacon Genomic Variation Query Standards" data-md-component="logo">
      
  <img src="../../img/GA-logo.png" alt="logo">

    </a>
    GA4GH Beacon Genomic Variation Query Standards
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/ga4gh-beacon/variant-query-types" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 480 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M186.1 328.7c0 20.9-10.9 55.1-36.7 55.1s-36.7-34.2-36.7-55.1 10.9-55.1 36.7-55.1 36.7 34.2 36.7 55.1M480 278.2c0 31.9-3.2 65.7-17.5 95-37.9 76.6-142.1 74.8-216.7 74.8-75.8 0-186.2 2.7-225.6-74.8-14.6-29-20.2-63.1-20.2-95 0-41.9 13.9-81.5 41.5-113.6-5.2-15.8-7.7-32.4-7.7-48.8 0-21.5 4.9-32.3 14.6-51.8 45.3 0 74.3 9 108.8 36 29-6.9 58.8-10 88.7-10 27 0 54.2 2.9 80.4 9.2 34-26.7 63-35.2 107.8-35.2 9.8 19.5 14.6 30.3 14.6 51.8 0 16.4-2.6 32.7-7.7 48.2 27.5 32.4 39 72.3 39 114.2m-64.3 50.5c0-43.9-26.7-82.6-73.5-82.6-18.9 0-37 3.4-56 6-14.9 2.3-29.8 3.2-45.1 3.2-15.2 0-30.1-.9-45.1-3.2-18.7-2.6-37-6-56-6-46.8 0-73.5 38.7-73.5 82.6 0 87.8 80.4 101.3 150.4 101.3h48.2c70.3 0 150.6-13.4 150.6-101.3m-82.6-55.1c-25.8 0-36.7 34.2-36.7 55.1s10.9 55.1 36.7 55.1 36.7-34.2 36.7-55.1-10.9-55.1-36.7-55.1"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../variant-scouts-home" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Introduction
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../requestProfiles_g_variant" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Beacon v2 Query Solutions
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../requestProfiles_VQSrequest" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Beacon v2+ VQS Proposals
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Query Parameter Definitions
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../variant-types" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Variation Types
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../genomic-coordinates-mappings" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Genomic Coordinates
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../terms-and-values" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Recommended Terms
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="https://github.com/ga4gh-beacon/variant-query-types/tree/main/src/requests" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Query Schema Source Files
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#adjacencyaccession" class="md-nav__link">
    <span class="md-ellipsis">
      AdjacencyAccession
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#adjacencystart" class="md-nav__link">
    <span class="md-ellipsis">
      AdjacencyStart
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#adjacencyend" class="md-nav__link">
    <span class="md-ellipsis">
      AdjacencyEnd
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#aminoacidchange" class="md-nav__link">
    <span class="md-ellipsis">
      AminoacidChange
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#copychange" class="md-nav__link">
    <span class="md-ellipsis">
      CopyChange
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#geneid" class="md-nav__link">
    <span class="md-ellipsis">
      GeneId
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#genomicfeature" class="md-nav__link">
    <span class="md-ellipsis">
      GenomicFeature
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#genomicalleleshortform" class="md-nav__link">
    <span class="md-ellipsis">
      GenomicAlleleShortForm
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#moleculareffect" class="md-nav__link">
    <span class="md-ellipsis">
      MolecularEffect
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#range" class="md-nav__link">
    <span class="md-ellipsis">
      Range
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#refgetaccession" class="md-nav__link">
    <span class="md-ellipsis">
      RefgetAccession
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#repeatsubunitlength" class="md-nav__link">
    <span class="md-ellipsis">
      RepeatSubunitLength
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#requestprofileid" class="md-nav__link">
    <span class="md-ellipsis">
      RequestProfileId
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sequence" class="md-nav__link">
    <span class="md-ellipsis">
      Sequence
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sequencelength" class="md-nav__link">
    <span class="md-ellipsis">
      SequenceLength
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sequencestart" class="md-nav__link">
    <span class="md-ellipsis">
      SequenceStart
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sequenceend" class="md-nav__link">
    <span class="md-ellipsis">
      SequenceEnd
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variantid" class="md-nav__link">
    <span class="md-ellipsis">
      VariantId
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#vrstype" class="md-nav__link">
    <span class="md-ellipsis">
      VRStype
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assembly" class="md-nav__link">
    <span class="md-ellipsis">
      Assembly
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#refseqid" class="md-nav__link">
    <span class="md-ellipsis">
      RefSeqId
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#referencebases" class="md-nav__link">
    <span class="md-ellipsis">
      ReferenceBases
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#alternatebases" class="md-nav__link">
    <span class="md-ellipsis">
      AlternateBases
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#varianttype" class="md-nav__link">
    <span class="md-ellipsis">
      VariantType
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#start" class="md-nav__link">
    <span class="md-ellipsis">
      Start
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#end" class="md-nav__link">
    <span class="md-ellipsis">
      End
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#matename" class="md-nav__link">
    <span class="md-ellipsis">
      MateName
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#matestart" class="md-nav__link">
    <span class="md-ellipsis">
      MateStart
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mateend" class="md-nav__link">
    <span class="md-ellipsis">
      MateEnd
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variantminlength" class="md-nav__link">
    <span class="md-ellipsis">
      VariantMinLength
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variantmaxlength" class="md-nav__link">
    <span class="md-ellipsis">
      VariantMaxLength
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="request-parameter-definitions">Request Parameter Definitions<a class="headerlink" href="#request-parameter-definitions" title="Permanent link">&para;</a></h1>
<p>Component definitions for <code>requestProfiles</code>. The definition of each parameter as a component allows for reuse across multiple request patterns but also for e.g. easy referencing in OpenAPI endpoints.</p>
<h2 id="adjacencyaccession"><code>AdjacencyAccession</code><a class="headerlink" href="#adjacencyaccession" title="Permanent link">&para;</a></h2>
<h4 id="description">Description<a class="headerlink" href="#description" title="Permanent link">&para;</a></h4>
<p>A GA4GH RefGet identifier for the <strong>adjacent</strong> sequence in adjacency/fusion scenarios.    </p>
<h4 id="definitions">Definitions<a class="headerlink" href="#definitions" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>VQS</code>        </li>
</ul>
</li>
<li>
<p><code>$ref</code>: <code>#/$defs/RefgetAccession</code>    </p>
</li>
</ul>
<h2 id="adjacencystart"><code>AdjacencyStart</code><a class="headerlink" href="#adjacencystart" title="Permanent link">&para;</a></h2>
<h4 id="description_1">Description<a class="headerlink" href="#description_1" title="Permanent link">&para;</a></h4>
<p>Start position or range (<em>i.e.</em> in case of a fuzzy breakpoint)<br />
on an adjacent sequence in case of a sequence adjecency      </p>
<p>Status:  </p>
<ul>
<li>PROPOSED FOR BEACON v2.n  </li>
<li>based on VRS v2 with modification    </li>
</ul>
<h4 id="definitions_1">Definitions<a class="headerlink" href="#definitions_1" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>VQS</code>        </li>
</ul>
</li>
<li>
<p><code>oneOf</code>:     </p>
<ul>
<li><code>integer</code>    </li>
<li><code>$ref: #/$defs/Range</code>        </li>
</ul>
</li>
</ul>
<h2 id="adjacencyend"><code>AdjacencyEnd</code><a class="headerlink" href="#adjacencyend" title="Permanent link">&para;</a></h2>
<h4 id="description_2">Description<a class="headerlink" href="#description_2" title="Permanent link">&para;</a></h4>
<p>End position or range (<em>i.e.</em> in case of a fuzzy breakpoint) on an<br />
adjacent sequence in case of a sequence adjecency      </p>
<p>Status:  </p>
<ul>
<li>PROPOSED FOR BEACON v2.n  </li>
<li>based on VRS v2 with modification    </li>
</ul>
<h4 id="definitions_2">Definitions<a class="headerlink" href="#definitions_2" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>VQS</code>        </li>
</ul>
</li>
<li>
<p><code>oneOf</code>:     </p>
<ul>
<li><code>integer</code>    </li>
<li><code>$ref: #/$defs/Range</code>        </li>
</ul>
</li>
</ul>
<h2 id="aminoacidchange"><code>AminoacidChange</code><a class="headerlink" href="#aminoacidchange" title="Permanent link">&para;</a></h2>
<h4 id="description_3">Description<a class="headerlink" href="#description_3" title="Permanent link">&para;</a></h4>
<p>Aminoacid alteration of interest in 1 letter format.      </p>
<p>Origin: Beacon v2.0    </p>
<h4 id="definitions_3">Definitions<a class="headerlink" href="#definitions_3" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>v2.0</code>    </li>
<li><code>VQS</code>        </li>
</ul>
</li>
<li>
<p><code>type</code>: <code>string</code>    </p>
</li>
<li>
<p><code>examples</code>:     </p>
<ul>
<li><code>V600E</code>    </li>
<li><code>M734V</code>        </li>
</ul>
</li>
</ul>
<h2 id="copychange"><code>CopyChange</code><a class="headerlink" href="#copychange" title="Permanent link">&para;</a></h2>
<h4 id="description_4">Description<a class="headerlink" href="#description_4" title="Permanent link">&para;</a></h4>
<p>MUST use a primaryCode representing one of:  </p>
<ul>
<li>EFO:0030069: complete genomic loss      </li>
<li>EFO:0020073: high-level loss      </li>
<li>EFO:0030068: low-level loss      </li>
<li>EFO:0030067: loss      </li>
<li>EFO:0030064: regional base ploidy      </li>
<li>EFO:0030070: gain      </li>
<li>EFO:0030071: low-level gain      </li>
<li>EFO:0030072: high-level gain    </li>
</ul>
<p>Endpoints are expected to provide query expansion according to the hierarchy<br />
of the terms:  </p>
<div class="highlight"><pre><span></span><code>- EFO:0030064  
- EFO:0030067  
    |- EFO:0030068  
    \- EFO:0020073  
         \- EFO:0030069  
- EFO:0030070  
    |- EFO:0030071  
    \- EFO:0030072  
</code></pre></div>
<p>Origin: Beacon v2.n, based on VRS v1.3+    </p>
<h4 id="definitions_4">Definitions<a class="headerlink" href="#definitions_4" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>VQS</code>        </li>
</ul>
</li>
<li>
<p><code>type</code>: <code>string</code>    </p>
</li>
<li>
<p><code>enum</code>:     </p>
<ul>
<li><code>EFO:0030069</code>    </li>
<li><code>EFO:0020073</code>    </li>
<li><code>EFO:0030068</code>    </li>
<li><code>EFO:0030067</code>    </li>
<li><code>EFO:0030064</code>    </li>
<li><code>EFO:0030070</code>    </li>
<li><code>EFO:0030071</code>    </li>
<li><code>EFO:0030072</code>        </li>
</ul>
</li>
</ul>
<h2 id="geneid"><code>GeneId</code><a class="headerlink" href="#geneid" title="Permanent link">&para;</a></h2>
<h4 id="description_5">Description<a class="headerlink" href="#description_5" title="Permanent link">&para;</a></h4>
<ul>
<li>A gene identifier  </li>
<li>It is strongly suggested to use a symbol following<br />
  the HGNC (<a href="https://www.genenames.org">www.genenames.org</a>) nomenclature.  </li>
</ul>
<p>Origin: Beacon v2.0    </p>
<h4 id="definitions_5">Definitions<a class="headerlink" href="#definitions_5" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>v2.0</code>    </li>
<li><code>VQS</code>        </li>
</ul>
</li>
<li>
<p><code>type</code>: <code>string</code>    </p>
</li>
<li>
<p><code>examples</code>:     </p>
<ul>
<li><code>BRAF</code>    </li>
<li><code>SCN5A</code>        </li>
</ul>
</li>
</ul>
<h2 id="genomicfeature"><code>GenomicFeature</code><a class="headerlink" href="#genomicfeature" title="Permanent link">&para;</a></h2>
<h4 id="description_6">Description<a class="headerlink" href="#description_6" title="Permanent link">&para;</a></h4>
<p>Descriptor of the class of genomic feature affected by the variant. Values<br />
from SO (Sequence ontology) are recommended, e.g.  </p>
<ul>
<li><code>SO:0001623</code> for 5 prime UTR variants  </li>
<li><code>SO:0000605</code> for intergenic variants  </li>
</ul>
<p>Note: While genes are features in SO they have a distinct parameter <code>geneId</code>.    </p>
<h4 id="definitions_6">Definitions<a class="headerlink" href="#definitions_6" title="Permanent link">&para;</a></h4>
<ul>
<li><code>type</code>: <code>string</code>    </li>
</ul>
<h2 id="genomicalleleshortform"><code>GenomicAlleleShortForm</code><a class="headerlink" href="#genomicalleleshortform" title="Permanent link">&para;</a></h2>
<h4 id="description_7">Description<a class="headerlink" href="#description_7" title="Permanent link">&para;</a></h4>
<p>HGVSId descriptor  </p>
<p>Origin: Beacon v2.0    </p>
<h4 id="definitions_7">Definitions<a class="headerlink" href="#definitions_7" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>v2.0</code>    </li>
<li><code>VQS</code>        </li>
</ul>
</li>
<li>
<p><code>type</code>: <code>string</code>    </p>
</li>
<li>
<p><code>examples</code>:     </p>
<ul>
<li><code>NM_004006.2:c.4375C&gt;T</code>        </li>
</ul>
</li>
</ul>
<h2 id="moleculareffect"><code>MolecularEffect</code><a class="headerlink" href="#moleculareffect" title="Permanent link">&para;</a></h2>
<h4 id="description_8">Description<a class="headerlink" href="#description_8" title="Permanent link">&para;</a></h4>
<p>Desscriptor of the molecular consequence generated by the variant. Use of<br />
Ontology terms from SO (Sequence Ontology) are recommended, e.g.   </p>
<ul>
<li><code>SO:0001583</code> for finding missense variants  </li>
<li><code>SO:0001580</code> for finding changes the coding sequence  </li>
<li><code>SO:0001587</code> for finding variants with a gained stop codon  </li>
</ul>
<p>The logical path for matching to the Beacon v2 model here would be <br />
<code>genomicVariations.molecularAttributes.molecularEffects.id</code>.    </p>
<h4 id="definitions_8">Definitions<a class="headerlink" href="#definitions_8" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>type</code>: <code>string</code>    </p>
</li>
<li>
<p><code>examples</code>:     </p>
<ul>
<li><code>SO:0001583</code>    </li>
<li><code>SO:0001580</code>    </li>
<li><code>SO:0001587</code>        </li>
</ul>
</li>
</ul>
<h2 id="range"><code>Range</code><a class="headerlink" href="#range" title="Permanent link">&para;</a></h2>
<h4 id="description_9">Description<a class="headerlink" href="#description_9" title="Permanent link">&para;</a></h4>
<p>An array of 2 integer values. If referring to sequence positions the<br />
"0-based, inclusive" format is used.    </p>
<h4 id="definitions_9">Definitions<a class="headerlink" href="#definitions_9" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>VQS</code>        </li>
</ul>
</li>
<li>
<p><code>type</code>: <code>array</code>    </p>
</li>
<li><code>items</code>:    <ul>
<li><code>type</code>: <code>integer</code>      </li>
<li><code>minimum</code>: <code>0</code>      </li>
<li><code>minItems</code>: <code>2</code>      </li>
<li><code>maxItems</code>: <code>2</code>    </li>
</ul>
</li>
</ul>
<h2 id="refgetaccession"><code>RefgetAccession</code><a class="headerlink" href="#refgetaccession" title="Permanent link">&para;</a></h2>
<h4 id="description_10">Description<a class="headerlink" href="#description_10" title="Permanent link">&para;</a></h4>
<p>A GA4GH RefGet identifier for the reference sequence, <em>i.e.</em> either a<br />
computed GA4GH checksum or other unique namespaced identifier supported<br />
by the server. It replaces the <code>referenceName</code> and <code>assemblyId</code> parameters<br />
according to the VRS v2 definition.  </p>
<p>Origin: VRS v2  </p>
<p>Reference: <a href="http://samtools.github.io/hts-specs/refget.html">GA4GH RefGet</a>    </p>
<h4 id="definitions_10">Definitions<a class="headerlink" href="#definitions_10" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>VQS</code>        </li>
</ul>
</li>
<li>
<p><code>type</code>: <code>string</code>    </p>
</li>
<li>
<p><code>examples</code>:     </p>
<ul>
<li><code>refseq:NC_000009.12</code>    </li>
<li><code>ga4gh:SQ.S_KjnFVz-FE7M0W6yoaUDgYxLPc1jyWU</code>        </li>
</ul>
</li>
</ul>
<h2 id="repeatsubunitlength"><code>RepeatSubunitLength</code><a class="headerlink" href="#repeatsubunitlength" title="Permanent link">&para;</a></h2>
<h4 id="description_11">Description<a class="headerlink" href="#description_11" title="Permanent link">&para;</a></h4>
<p>The number of residues in a repeat subunit. In contrast to the VRS model we<br />
allow for range queries (e.g. all repeats with subunits of 1 or 2).  </p>
<p>Origin: VRS v2    </p>
<h4 id="definitions_11">Definitions<a class="headerlink" href="#definitions_11" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>VQS</code>        </li>
</ul>
</li>
<li>
<p><code>oneOf</code>:     </p>
<ul>
<li><code>integer</code>    </li>
<li><code>$ref: #/$defs/Range</code>        </li>
</ul>
</li>
</ul>
<h2 id="requestprofileid"><code>RequestProfileId</code><a class="headerlink" href="#requestprofileid" title="Permanent link">&para;</a></h2>
<h4 id="description_12">Description<a class="headerlink" href="#description_12" title="Permanent link">&para;</a></h4>
<p>The <code>requestProfile</code> parameter here allows beacons to check the type of query<br />
being performed and to compare the provided request parameters for conformity<br />
with the expected query profile.<br />
The parameter definition here is a placeholder; the definitions of typed queries<br />
will use constant values for their <code>requestProfile</code> parameter.  </p>
<p>Origin: Beacon v2+    </p>
<h4 id="definitions_12">Definitions<a class="headerlink" href="#definitions_12" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>VQS</code>        </li>
</ul>
</li>
<li>
<p><code>type</code>: <code>string</code>    </p>
</li>
</ul>
<h2 id="sequence"><code>Sequence</code><a class="headerlink" href="#sequence" title="Permanent link">&para;</a></h2>
<h4 id="description_13">Description<a class="headerlink" href="#description_13" title="Permanent link">&para;</a></h4>
<p>DNA bases.      </p>
<ul>
<li>Accepted values: <code>[ACGTN]*</code>      </li>
<li>N is a wildcard, that denotes the position of any base,<br />
  and can be used as a standalone base of any type or within a partially known<br />
  sequence. As example, a query of <code>ANNT</code> the Ns can take take any form of [ACGT]<br />
  and will match <code>ANNT</code>, <code>ACNT</code>, <code>ACCT</code>, <code>ACGT</code> ... and so forth.      </li>
</ul>
<p>Origin: VRS v1.n  </p>
<p>TODO: Review use of base characters.    </p>
<h4 id="definitions_13">Definitions<a class="headerlink" href="#definitions_13" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>VQS</code>        </li>
</ul>
</li>
<li>
<p><code>type</code>: <code>string</code>    </p>
</li>
</ul>
<h2 id="sequencelength"><code>SequenceLength</code><a class="headerlink" href="#sequencelength" title="Permanent link">&para;</a></h2>
<h4 id="description_14">Description<a class="headerlink" href="#description_14" title="Permanent link">&para;</a></h4>
<p>The sequence length when querying the conceptual representation of a<br />
sequence according to a VRS <code>ReferenceLengthExpression</code> class.  </p>
<p>Additionally, the <code>sequenceLength</code> parameter can be used to limit the length<br />
of matched variants, e.g. by specifying a range of lengths for <code>copyChange</code><br />
matches.  </p>
<p>Origin: VRS v2    </p>
<h4 id="definitions_14">Definitions<a class="headerlink" href="#definitions_14" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>VQS</code>        </li>
</ul>
</li>
<li>
<p><code>oneOf</code>:     </p>
<ul>
<li><code>integer</code>    </li>
<li><code>$ref: #/$defs/Range</code>        </li>
</ul>
</li>
</ul>
<h2 id="sequencestart"><code>SequenceStart</code><a class="headerlink" href="#sequencestart" title="Permanent link">&para;</a></h2>
<h4 id="description_15">Description<a class="headerlink" href="#description_15" title="Permanent link">&para;</a></h4>
<p>Start position or range (<em>i.e.</em> in case of a fuzzy breakpoint) on a sequence.<br />
Status: PROPOSED FOR BEACON v2.n, based on VRS v2 with modification    </p>
<h4 id="definitions_15">Definitions<a class="headerlink" href="#definitions_15" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>VQS</code>        </li>
</ul>
</li>
<li>
<p><code>oneOf</code>:     </p>
<ul>
<li><code>integer</code>    </li>
<li><code>$ref: #/$defs/Range</code>        </li>
</ul>
</li>
</ul>
<h2 id="sequenceend"><code>SequenceEnd</code><a class="headerlink" href="#sequenceend" title="Permanent link">&para;</a></h2>
<h4 id="description_16">Description<a class="headerlink" href="#description_16" title="Permanent link">&para;</a></h4>
<p>End position or range (<em>i.e.</em> in case of a fuzzy breakpoint) on a sequence.<br />
Status: PROPOSED FOR BEACON v2.n, based on VRS v2 with modification    </p>
<h4 id="definitions_16">Definitions<a class="headerlink" href="#definitions_16" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>VQS</code>        </li>
</ul>
</li>
<li>
<p><code>oneOf</code>:     </p>
<ul>
<li><code>integer</code>    </li>
<li><code>$ref: #/$defs/Range</code>        </li>
</ul>
</li>
</ul>
<h2 id="variantid"><code>VariantId</code><a class="headerlink" href="#variantid" title="Permanent link">&para;</a></h2>
<h4 id="description_17">Description<a class="headerlink" href="#description_17" title="Permanent link">&para;</a></h4>
<ul>
<li>A variant identifier such as a VRSid, ClinVar id, dbSNP rsID or a<br />
  COSMIC identifier  </li>
<li>In the default data model this query parameter corresponds to<br />
<code>identifiers.variantAlternateIds</code> but potentially can map to other<br />
  identifiers as well. However, no specific query parameterwas defined<br />
  in the Beacon v2 default model.<br />
Status: PROPOSED FOR BEACON v2.n    </li>
</ul>
<h4 id="definitions_17">Definitions<a class="headerlink" href="#definitions_17" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>VQS</code>        </li>
</ul>
</li>
<li>
<p><code>type</code>: <code>string</code>    </p>
</li>
<li>
<p><code>examples</code>:     </p>
<ul>
<li><code>ClinGen:CA152954</code>    </li>
<li><code>dbSNP:rs587780345</code>        </li>
</ul>
</li>
</ul>
<h2 id="vrstype"><code>VRStype</code><a class="headerlink" href="#vrstype" title="Permanent link">&para;</a></h2>
<h4 id="description_18">Description<a class="headerlink" href="#description_18" title="Permanent link">&para;</a></h4>
<p>Type of the variation according to the VRS model. Examples are here e.g.<br />
<code>Adjacency</code> or <code>Allele</code>.<br />
Origin: VRS v2    </p>
<h4 id="definitions_18">Definitions<a class="headerlink" href="#definitions_18" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>VQS</code>        </li>
</ul>
</li>
<li>
<p><code>type</code>: <code>string</code>    </p>
</li>
<li>
<p><code>enum</code>:     </p>
<ul>
<li><code>Adjacency</code>    </li>
<li><code>Allele</code>    </li>
<li><code>CisPhasedBlock</code>    </li>
<li><code>CopyNumberChange</code>    </li>
<li><code>CopyNumberCount</code>    </li>
<li><code>DerivativeMolecule</code>    </li>
<li><code>Terminus</code>        </li>
</ul>
</li>
</ul>
<h2 id="assembly"><code>Assembly</code><a class="headerlink" href="#assembly" title="Permanent link">&para;</a></h2>
<h4 id="description_19">Description<a class="headerlink" href="#description_19" title="Permanent link">&para;</a></h4>
<p>Genomic assembly accession and version as RefSqeq assembly accession<br />
(e.g. "GCF_000001405.39") or a versioned assembly name or synonym such as UCSC<br />
Genome Browser assembly (e.g. "hg38") or Genome Reference Consortium Human (e.g.<br />
"GRCh38.p13") names.<br />
DEPRECATION NOTE: The use of a assembly specific sequence identifier obviates<br />
this parameter. Not part of VRS v2 aligned model versions.    </p>
<h4 id="definitions_19">Definitions<a class="headerlink" href="#definitions_19" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>v1</code>    </li>
<li><code>v2.0</code>    </li>
<li><code>v2.1</code>        </li>
</ul>
</li>
<li>
<p><code>type</code>: <code>string</code>    </p>
</li>
<li>
<p><code>example</code>:     </p>
<ul>
<li><code>GCF_000001405.39</code>    </li>
<li><code>hg38</code>    </li>
<li><code>GRCh38.p13</code>        </li>
</ul>
</li>
</ul>
<h2 id="refseqid"><code>RefSeqId</code><a class="headerlink" href="#refseqid" title="Permanent link">&para;</a></h2>
<h4 id="description_20">Description<a class="headerlink" href="#description_20" title="Permanent link">&para;</a></h4>
<p>Reference sequence id for genomic reference sequence in which variant<br />
coordinates are given, e.g. "refseq:NC_000009.12" for human chromosome 9 in the<br />
GRCh38 assembly. The use of the assembly specific RefSeqId is recommended<br />
although alternatively names, synonymous or aliases e.g. "chr9" could be<br />
used in conjunction with an <code>Assembly</code> parameter.<br />
DEPRECATION NOTE: To be replaced with the <code>RefgetAccession</code> from<br />
VRS v2.    </p>
<h4 id="definitions_20">Definitions<a class="headerlink" href="#definitions_20" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>v1</code>    </li>
<li><code>v2.0</code>    </li>
<li><code>v2.1</code>        </li>
</ul>
</li>
<li>
<p><code>type</code>: <code>string</code>    </p>
</li>
<li>
<p><code>example</code>:     </p>
<ul>
<li><code>refseq:NC_000009.12</code>    </li>
<li><code>chr9</code>    </li>
<li><code>NC_012920.1</code>        </li>
</ul>
</li>
</ul>
<h2 id="referencebases"><code>ReferenceBases</code><a class="headerlink" href="#referencebases" title="Permanent link">&para;</a></h2>
<h4 id="description_21">Description<a class="headerlink" href="#description_21" title="Permanent link">&para;</a></h4>
<p>The reference bases for the variant at the indicated position. It is based<br />
on the VCF concept of having (anchored) reference bases at an indicated<br />
genomic location in combination with <code>alternateBases</code> to define their<br />
replacement. In contrast, standards such as GA4GH VRS only indicate the<br />
<code>sequence</code> observed at a given base position, including the use of an empty<br />
sequence together with <code>start</code> + <code>end</code> positions with <code>end - start &gt; 0</code><br />
to indicate deletions.<br />
Origin: VCF derived (optional) use in Beacon v0.3 -&gt; v2.1<br />
Status: LEGACY    </p>
<h4 id="definitions_21">Definitions<a class="headerlink" href="#definitions_21" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>v1</code>    </li>
<li><code>v2.0</code>    </li>
<li><code>v2.1</code>        </li>
</ul>
</li>
<li>
<p><code>$ref</code>: <code>#/$defs/Sequence</code>    </p>
</li>
</ul>
<h2 id="alternatebases"><code>AlternateBases</code><a class="headerlink" href="#alternatebases" title="Permanent link">&para;</a></h2>
<h4 id="description_22">Description<a class="headerlink" href="#description_22" title="Permanent link">&para;</a></h4>
<p>The bases of a sequence variant at a given position differing from the<br />
reference sequence, as defined by the <code>referenceBases</code> parameter. Please<br />
see <code>refereenceBases</code> for further information.<br />
Origin: VCF derived use in Beacon v0.3 -&gt; v2.1<br />
Status: LEGACY    </p>
<h4 id="definitions_22">Definitions<a class="headerlink" href="#definitions_22" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>v1</code>    </li>
<li><code>v2.0</code>    </li>
<li><code>v2.1</code>        </li>
</ul>
</li>
<li>
<p><code>$ref</code>: <code>#/$defs/Sequence</code>    </p>
</li>
</ul>
<h2 id="varianttype"><code>VariantType</code><a class="headerlink" href="#varianttype" title="Permanent link">&para;</a></h2>
<h4 id="description_23">Description<a class="headerlink" href="#description_23" title="Permanent link">&para;</a></h4>
<p>The <code>variantType</code> is used to query variants which are not defined through<br />
a sequence of one or more bases using the <code>alternateBases</code> parameter. This<br />
VCF derived parameter is being replaced by the more specific VRS derived<br />
parameters such as <code>copyChange</code>.<br />
(Legacy) Examples here are e.g. structural variants:<br />
<em> DUP<br />
  - increased allelic count of material from the genomic region between<br />
<code>start</code> and <code>end</code> positions<br />
  - no assumption about the placement of the additional sequences is being<br />
    made (i.e. no <em>in situ</em> requirement as tandem duplications)<br />
</em> DEL: deletion of sequence following <code>start</code>  </p>
<p>In contrast to the updated VRS based v2.n parameters such as <code>copyChange</code><br />
the Beacon v1.1 -&gt; v2.1 query model is not prescriptive with regard to the<br />
values allowed for <code>variantType</code> with use of extended types (e.g. <code>EFO:0030063</code>)<br />
being permitted. However, a support for the basic CNV types above - where<br />
represented in the data - is recommended.<br />
Status: LEGACY with potential use in v2.n for non-CNV parameters <br />
Note: The VRS v2 <code>copyChange</code> is now a partial and more specific replacement<br />
      over <code>variantType</code> for copy number variations. However, additional<br />
      concepts so far have not been covered and might warrant use of an<br />
      additional parameter (<code>variantClass</code>?).    </p>
<h4 id="definitions_23">Definitions<a class="headerlink" href="#definitions_23" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>v1</code>    </li>
<li><code>v2.0</code>    </li>
<li><code>v2.1</code>        </li>
</ul>
</li>
<li>
<p><code>type</code>: <code>string</code>    </p>
</li>
<li>
<p><code>examples</code>:     </p>
<ul>
<li><code>EFO:0030070</code>    </li>
<li><code>DUP</code>    </li>
<li><code>DEL</code>    </li>
<li><code>EFO:0030069</code>        </li>
</ul>
</li>
</ul>
<h2 id="start"><code>Start</code><a class="headerlink" href="#start" title="Permanent link">&para;</a></h2>
<h4 id="description_24">Description<a class="headerlink" href="#description_24" title="Permanent link">&para;</a></h4>
<p>NOTE: This parameter will be <em>potentially</em> replaced by the VRS based definition<br />
      which uses either an integer or a Range (2 integers) in contrast to<br />
      the use of an array with 1 or 2 integers here. The difference lies in <br />
      the format of "1 integer array" versus "1 integer".<br />
Precise or fuzzy start coordinate position(s), allele locus<br />
(0-based, inclusive).  </p>
<ul>
<li><code>start</code> only:  </li>
<li>for single positions, e.g. the start of a specified sequence<br />
    alteration where the size is given through the specified <code>alternateBases</code>  </li>
<li>typical use are queries for SNV and small InDels  </li>
<li>the use of <code>start</code> without an <code>end</code> parameter requires the use of<br />
<code>alternateBases</code>  </li>
<li>1 value in both <code>start</code> and <code>end</code>:  </li>
<li>for searching any variant falling fully or partially within the range<br />
    between <code>start</code> and <code>end</code> (a.k.a. "range query")  </li>
<li>additional use of <code>variantType</code> OR <code>alternateBases</code> can limit the<br />
    scope of the query  </li>
<li>by convention, partial overlaps of variants with the indicated genomic<br />
    range are accepted; for specific overlap requirements the 4-parameter<br />
    "Bracket Queries" should be employed  </li>
<li>2 values in both <code>start</code> and <code>end</code> for constructing a "Bracket Query":  </li>
<li>can be used to match any contiguous genomic interval, e.g. for querying<br />
    imprecise positions  </li>
<li>identifies all structural variants starting between <code>start[0]</code> and <code>start[1]</code>,<br />
    and ending between <code>end[0]</code> &lt;-&gt; <code>end[1]</code>  </li>
<li>single or double sided precise matches can be achieved by setting<br />
<code>start[1]=start[0]+1</code> and <code>end[1]=end[0]+1</code>    </li>
</ul>
<h4 id="definitions_24">Definitions<a class="headerlink" href="#definitions_24" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>v2.0</code>    </li>
<li><code>v2.1</code>        </li>
</ul>
</li>
<li>
<p><code>type</code>: <code>array</code>    </p>
</li>
<li>
<p><code>items</code>:    </p>
<ul>
<li><code>type</code>: <code>integer</code>      </li>
<li><code>format</code>: <code>int64</code>      </li>
<li><code>minimum</code>: <code>0</code>    </li>
</ul>
</li>
<li>
<p><code>minItems</code>: <code>1</code>    </p>
</li>
<li>
<p><code>maxItems</code>: <code>2</code>    </p>
</li>
</ul>
<h2 id="end"><code>End</code><a class="headerlink" href="#end" title="Permanent link">&para;</a></h2>
<h4 id="description_25">Description<a class="headerlink" href="#description_25" title="Permanent link">&para;</a></h4>
<h5 id="notes">Notes<a class="headerlink" href="#notes" title="Permanent link">&para;</a></h5>
<p>See the <code>start</code> parameter for information on the potential replacement<br />
of this parameter with the VRS based definition.  </p>
<p>Precise or bracketing the end of the variants of interest:  </p>
<ul>
<li>(0-based, exclusive) - see <code>start</code>  </li>
<li>for bracket queries, provide 2 values (e.g. [111,222]).    </li>
</ul>
<h4 id="definitions_25">Definitions<a class="headerlink" href="#definitions_25" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>v2.0</code>    </li>
<li><code>v2.1</code>        </li>
</ul>
</li>
<li>
<p><code>type</code>: <code>array</code>    </p>
</li>
<li>
<p><code>items</code>:    </p>
<ul>
<li><code>type</code>: <code>integer</code>      </li>
<li><code>format</code>: <code>int64</code>      </li>
<li><code>minimum</code>: <code>1</code>    </li>
</ul>
</li>
<li>
<p><code>minItems</code>: <code>1</code>    </p>
</li>
<li>
<p><code>maxItems</code>: <code>2</code>    </p>
</li>
</ul>
<h2 id="matename"><code>MateName</code><a class="headerlink" href="#matename" title="Permanent link">&para;</a></h2>
<h4 id="description_26">Description<a class="headerlink" href="#description_26" title="Permanent link">&para;</a></h4>
<h5 id="notes_1">Notes<a class="headerlink" href="#notes_1" title="Permanent link">&para;</a></h5>
<ul>
<li>while the <code>mateName</code> parameter was originally defined for Beacon v1.1<br />
  it was never properly documented and is not considered a part of the<br />
  supported Beacon v2.n specification. It is now fully implemented in the<br />
  VRS v2 based <code>adjacencyAccession</code> parameter.  </li>
</ul>
<p>Status: DEPRECATED in v2.n    </p>
<h4 id="definitions_26">Definitions<a class="headerlink" href="#definitions_26" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>v1.1</code>    </li>
<li><code>v2.0</code>    </li>
<li><code>v2.1</code>        </li>
</ul>
</li>
<li>
<p><code>$ref</code>: <code>#/$defs/RefSeqId</code>    </p>
</li>
</ul>
<h2 id="matestart"><code>MateStart</code><a class="headerlink" href="#matestart" title="Permanent link">&para;</a></h2>
<h4 id="description_27">Description<a class="headerlink" href="#description_27" title="Permanent link">&para;</a></h4>
<p>genomic start position of fusion partner breakpoint region<br />
Status: DEPRECATED in v2.n (see <code>mateName</code>)    </p>
<h4 id="definitions_27">Definitions<a class="headerlink" href="#definitions_27" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>v1.1</code>    </li>
<li><code>v2.0</code>    </li>
<li><code>v2.1</code>        </li>
</ul>
</li>
<li>
<p><code>type</code>: <code>integer</code>    </p>
</li>
</ul>
<h2 id="mateend"><code>MateEnd</code><a class="headerlink" href="#mateend" title="Permanent link">&para;</a></h2>
<h4 id="description_28">Description<a class="headerlink" href="#description_28" title="Permanent link">&para;</a></h4>
<p>genomic end position of fusion partner breakpoint region<br />
Status: DEPRECATED in v2.n (see <code>mateName</code>)    </p>
<h4 id="definitions_28">Definitions<a class="headerlink" href="#definitions_28" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>v1.1</code>    </li>
<li><code>v2.0</code>    </li>
<li><code>v2.1</code>        </li>
</ul>
</li>
<li>
<p><code>type</code>: <code>integer</code>    </p>
</li>
</ul>
<h2 id="variantminlength"><code>VariantMinLength</code><a class="headerlink" href="#variantminlength" title="Permanent link">&para;</a></h2>
<h4 id="description_29">Description<a class="headerlink" href="#description_29" title="Permanent link">&para;</a></h4>
<ul>
<li>Minimum length in bases of a genomic variant  </li>
<li>This is an optional parameter without prescribed use. While a length is<br />
  commonly available for structural variants such as copy number variations,<br />
  it is recommended that length based queries should also be supported for<br />
  variants with indicated referenceBases and alternateBases, to enable<br />
  length-specific wildcard queries.  </li>
</ul>
<p>Origin: Beacon v2.0  </p>
<p>Status: DEPRECATED in v2.n (see <code>sequenceLength</code>)    </p>
<h4 id="definitions_29">Definitions<a class="headerlink" href="#definitions_29" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>v2.0</code>    </li>
<li><code>v2.1</code>        </li>
</ul>
</li>
<li>
<p><code>type</code>: <code>integer</code>    </p>
</li>
<li>
<p><code>format</code>: <code>int64</code>    </p>
</li>
<li>
<p><code>minimum</code>: <code>0</code>    </p>
</li>
</ul>
<h2 id="variantmaxlength"><code>VariantMaxLength</code><a class="headerlink" href="#variantmaxlength" title="Permanent link">&para;</a></h2>
<h4 id="description_30">Description<a class="headerlink" href="#description_30" title="Permanent link">&para;</a></h4>
<ul>
<li>Maximum length in bases of a genomic variant.  </li>
<li>This is an optional parameter without prescribed use. While a length is<br />
  commonly available for structural variants such as copy number variations,<br />
  it is recommended that length based queries should also be supported for<br />
  variants with indicated referenceBases and alternateBases, to enable<br />
  length-specific wildcard queries.  </li>
</ul>
<p>Status: DEPRECATED in v2.n (see <code>sequenceLength</code>)  </p>
<p>Origin: Beacon v2.0    </p>
<h4 id="definitions_30">Definitions<a class="headerlink" href="#definitions_30" title="Permanent link">&para;</a></h4>
<ul>
<li>
<p><code>versions</code>:     </p>
<ul>
<li><code>v2.0</code>    </li>
<li><code>v2.1</code>        </li>
</ul>
</li>
<li>
<p><code>type</code>: <code>integer</code>    </p>
</li>
<li>
<p><code>format</code>: <code>int64</code>    </p>
</li>
<li>
<p><code>minimum</code>: <code>1</code>    </p>
</li>
</ul>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      &copy; Copyright 2024-2025 GA4GH and Beacon Contributors
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": ["content.tabs.link", "search.highlight", "search.share", "navigation.instant", "navigation.tracking", "navigation.sections"], "search": "../../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.13a4f30d.min.js"></script>
      
    
  </body>
</html>